lms/djangoapps/mobile_api/video_outlines/serializers.py

```diff
@@ -159,23 +159,37 @@ def find_urls(course_id, block, child_to_parent, request):
     return unit_url, section_url
 
 
-def video_summary(course, course_id, video_descriptor, request, local_cache):
+def video_summary(video_profiles, course_id, video_descriptor, request, local_cache):
     """
     returns summary dict for the given video module
     """
-    # First try to check VAL for the URLs we want.
-    val_video_info = local_cache['course_videos'].get(video_descriptor.edx_video_id, {})
-    if val_video_info:
-        video_url = val_video_info['url']
+    # Get encoded videos
+    encoded_videos = local_cache['course_videos'].get(video_descriptor.edx_video_id, {})
+
+    # Get highest priority video to populate backwards compatible field
+    default_encoded_video = None
+    if encoded_videos:
+        for profile in video_profiles:
```

Comment:
have you verified that the order of the profiles in `video_profiles` is retained?  That is, we should make sure that `mobile_low` is given precedence over `mobile_high` when finding the `default_encoded_video`.